### ISSUE TITLE
ubuntu gets stuck when running

### ISSUE THREAD

**Issue by ody999** [utmapp/UTM#5205]
**Describe the issue**
When running ubuntu20.04.5, chances are that the system gets totally stuck, which means I can only move the mouse cursor, and the system no longer responses to any keyboard or mouse click input. Does this has something to do with display cards with '-gl', since when choosing cards that do not support GPU acceleration, so far haven't met this situation. 

**Configuration (required)**
* UTM Version: v4.1.6
* OS Version: ipados 15.4.1
* Device Model: ipadpro2021
* Is it jailbroken (name jailbreak used)? no 
* How did you install UTM?  trollstore

**Crash log**
n/a

**Debug log**
n/a

**Upload VM**
n/a

**Comment by ody999** [comment]
interestingly，when the screen display stagnates, the website background music is still able to play.

**Comment by jmzwcn** [comment]
Yes, I have hit the same issue too, and have to reboot without any saved.

**Comment by thedarb** [comment]
Same. Without any GL video options, the I can make it randomly from a few hours to up to around 4 days before I get such a hang.  When it finally does, it's the same.  Audio keeps playing, and I can ssh into the vm.  But attempting to kill and restart X or the dm completes the freeze, and nothing works.

I have the problem in X11 Cinammon, X11 KDE, and Wayland KDE.  Running Ubuntu 22.04.  Don't know about gnome desktop, I won't install that.

**Comment by adanowitz** [comment]
Same problem running an MBP 16" with M2. Ubuntu Desktop installed. Using Apple's virtualization back-end rather than qemu.

**Comment by stuaxo** [comment]
I made some progress - I tried looking into resetting the graphics device.
I didn't manage that directly, but I did manage to disconnect and reconnect it - this *does* log out you, but you are back a the GUI and you can either log back in and continue, or you can cleanly shut down.



As root, run lspci to find theVirtio GPU device, then look at the device number on the left
```
lspci
00:00.0 Host bridge: Red Hat, Inc. QEMU PCIe Host bridge
00:01.0 Ethernet controller: Red Hat, Inc. Virtio network device
00:02.0 Ethernet controller: Red Hat, Inc. Virtio network device
00:03.0 Display controller: Red Hat, Inc. Virtio GPU (rev 01)
00:04.0 USB controller: NEC Corporation uPD720200 USB 3.0 Host Controller (rev 03)
00:05.0 USB controller: Red Hat, Inc. QEMU XHCI Host Controller (rev 01)
00:06.0 SCSI storage controller: Red Hat, Inc. Virtio block device
00:07.0 Communication controller: Red Hat, Inc. Virtio console
00:08.0 Unclassified device [0002]: Red Hat, Inc. Virtio filesystem
00:09.0 Unclassified device [00ff]: Red Hat, Inc. Virtio RNG
00:0a.0 Unclassified device [00ff]: Red Hat, Inc. Virtio memory balloon
```

Here it's 00:03.0

Remove the virtual gpu
```
echo 1 > /sys/bus/pci/devices/0000\:00\:03.0/remove
```

Rescan
```
 echo "1" > /sys/bus/pci/rescan
```
Without this bit the folder under /sys/bus/pci/devices/0000\:00\:03.0/ no longer exists.

At this point you may be at the login screen straight away, but I wasn't - I got there by getting gdm3 to restart
```
systemctl restart gdm3
```

An interesting thing is that performing the PCI removal step while things are running normally reproduces the bug I see exactly - mouse cursor moves but no interaction otherwise.

Instead of restarting gdm3 we might be able to do something with more finesse and actually get the desktop back, there is also currently work in gnome to survive GPU resets, once this lands that could help.
Being able to reset the GPU directly would also help (a /reset node in the device directory)

**Comment by osy** [comment]
I think I found the root cause https://github.com/utmapp/CocoaSpice/issues/5
